File: PITA-Taller1/cpp/src/Faculty.cpp
// Initial implementation of the Faculty class.
// TODO: define faculty behavior as system logic is added.

#include "Faculty.h"

Faculty::Faculty()
    : faculty_id(0),
      name(""),
      dean(""),
      creation_date(""),
      active(false),
      program_list() {}

Faculty::Faculty(int faculty_id,
                   const std::string& name,
                   const std::string& dean,
                   const std::string& creation_date,
                   bool active,
                   const LinkedList<Program*>& program_list)
    : faculty_id(faculty_id),
      name(name),
      dean(dean),
      creation_date(creation_date),
      active(active),
      program_list(program_list) {}

int Faculty::get_faculty_id() const { return faculty_id; }
std::string Faculty::get_name() const { return name; }
void Faculty::set_name(const std::string& value) { name = value; }
std::string Faculty::get_dean() const { return dean; }
void Faculty::set_dean(const std::string& value) { dean = value; }
std::string Faculty::get_creation_date() const { return creation_date; }
void Faculty::set_creation_date(const std::string& value) { creation_date = value; }
bool Faculty::is_active() const { return active; }
void Faculty::set_active(bool value) { active = value; }
LinkedList<Program*>& Faculty::get_program_list() { return program_list; }
const LinkedList<Program*>& Faculty::get_program_list() const { return program_list; }

File: PITA-Taller1/python/models/faculty.py
"""Faculty model.

Represents a university faculty and stores the related academic programs.
"""

from .linked_list import LinkedList


class Faculty:
    """Entity that represents a university faculty."""

    def __init__(
        self,
        faculty_id=0,
        name="",
        dean="",
        creation_date="",
        active=False,
        program_list=None,
    ):
        self.faculty_id = faculty_id
        self.name = name
        self.dean = dean
        self.creation_date = creation_date
        self.active = active
        self.program_list = program_list if program_list is not None else LinkedList()
